LvL implementation

tomadror opened the pull request:
fLvL.py is a function that computes the LvL organization score of a given 2D cloud field given by its cloud mask.

tomadror committed:
LvL implementation 

fLvL.py is a function that computes the LvL organization score of a given 2D cloud field given by its cloud mask.

martinjanssens commented:
Fantastic!! Thanks a lot for doing this so, so quickly :) Just a few points to think about (@leifdenby feel free to chime in too):
- If the function `lvl()` operates directly on a binary cloud mask, can we put it in the `cloudmetrics/mask` directory? I think this makes sense, even though the metric strictly speaking is object-based, because you work on 1D objects, and not 2D objects (like the rest of our object metrics). Then you also don't have to use the rest of our object identification routines, and you can use the ones you prefer (and currently use).
- Which packages are needed to calculate the metric, which we currently don't have in our list of requirements (https://github.com/cloudsci/cloudmetrics/blob/master/setup.cfg)? `PIL` and `cv2`?
- Could you add a small piece of documentation atop the implemented LvL function (look at `fractal_dimension` as an example), maybe including a reference to your paper? And perhaps also work with the same field names as in the rest of the package (i.e. `cloud_mask` instead of `A`)?
- To be consistent with the filenaming convention, I wonder how we should call the file. Perhaps `lvl.py`?
- To be sure that the function is implemented the way you want it to, and that future changes we make to the repository don't mess it up, could you implement a small test for how it is supposed to work in `tests/test_lvl.py`? See e.g. the example from `test_cloud_fraction.py` for how to do this. The example could e.g. be for the Bernoulli random field from your paper, and perhaps two examples where either L is supposed to be large?
- You can test whether your implementation will be mergeable with our main branch by installing pre-commit (https://github.com/cloudsci/cloudmetrics/blob/master/docs/developing.md); this will also test whether the typesetting of the code adheres to our standard, and fix it where it doesn't.

tomadror committed:
Update fLVL.py

Renaming the field name from "A" to "cloud_mask."
Renaming the function from "fLVL.py" to "LvL.py."

tomadror committed:
Update and rename fLVL.py to LvL.py

Change name of file to "LvL.py"
